dracut/30ignition/persist-ifcfg.sh

```diff
@@ -33,6 +35,32 @@ persist_ifcfg() {
         return 0
     fi
 
+    # Persist the hostname if the admin has elected to use the dracut method of
+    # defining the IP on the commandline, but only if Ignition hasn't set the hostname.
+    hpath="/sysroot/etc/hostname"
+    hname=$(< /proc/sys/kernel/hostname)
+    for iface in $(ls /sys/class/net)
+    do
+        # Find the totems indicating that dracut set up the interface
+        iface_totems="/run/initramfs/net.${iface}"
+        [ -f "${iface_totems}.did-setup" ] || continue;
+        [ -f "${iface_totems}.hostname" ] || continue;
+
+        # The format of the file is _the command used to set the hostname_
+        # echo hostname > /proc/sys/kernel/hostname.
+        read _ iface_hostname _ < "${iface_totems}.hostname"
+
+        # Ensure that the hostname used by the kernel is the same one
+        # used by this interface. This guard is intended to protect against
+        # mixed `ip=` or where one is set via dhcp.
+        if [ "${iface_hostname}" == "${hname}" ] && [ ! -f "${hpath}" ]; then
```

Comment:
Hmm, would be cleaner to do the `[ ! -f "${hpath}" ]` check upfront so we can avoid entering the loop entirely, but not a blocker!

Reply:
Yeah, I was trying to avoid the nested if's, and figured the cost of the loop was trivial. I agree, though.